src/linker/Linker.Dataflow/XmlFlowAnnotationSource.cs

```diff
@@ -0,0 +1,690 @@
+			ArrayBuilder<Attribute> attributes = ProcessAttributes (iterator.Current.SelectChildren ("attribute", _ns));
+			ArrayBuilder<(string, ArrayBuilder<Attribute>)> parameterAnnotations = ProcessParameters (type,
+				method, iterator.Current.SelectChildren ("parameter", _ns));
+			ArrayBuilder<ArrayBuilder<Attribute>> returnParameterAnnotations = ProcessReturnParameters (type,
+				method, iterator.Current.SelectChildren ("returnparameter", _ns));
+
+			var parameterAnnotation = new ArrayBuilder<(string ParamName, DynamicallyAccessedMemberTypes Annotation)> ();
+			DynamicallyAccessedMemberTypes returnAnnotation = 0;
+
+			if (parameterAnnotations.Count > 0) {
+				foreach (var parameter in parameterAnnotations.ToArray ()) {
+					DynamicallyAccessedMemberTypes paramAnnotation = ParseKinds (parameter.Item2);
+					if (paramAnnotation != 0)
+						parameterAnnotation.Add ((parameter.Item1, paramAnnotation));
+				}
+			}
+
+			if (returnParameterAnnotations.Count == 1) {
+				foreach (var returnparameter in returnParameterAnnotations.ToArray ()) {
+					DynamicallyAccessedMemberTypes returnparamAnnotation = ParseKinds (returnparameter);
+					if (returnparamAnnotation != 0)
+						returnAnnotation = returnparamAnnotation;
+				}
+			}
+			if (returnAnnotation != 0 || parameterAnnotation.Count > 0)
+				_methods[method] = new AnnotatedMethod (returnAnnotation, parameterAnnotation.ToArray ());
+
+		}
+
+		ArrayBuilder<(string, ArrayBuilder<Attribute>)> ProcessParameters (TypeDefinition type,
+			MethodDefinition method, XPathNodeIterator iterator)
+		{
+			var methodParameters = new ArrayBuilder<(string, ArrayBuilder<Attribute>)> ();
+			while (iterator.MoveNext ()) {
+				methodParameters.Add ((GetAttribute (iterator.Current, "name"), ProcessAttributes (iterator.Current.SelectChildren ("attribute", _ns))));
+			}
+			return methodParameters;
+		}
+
+		ArrayBuilder<ArrayBuilder<Attribute>> ProcessReturnParameters (TypeDefinition type,
+			MethodDefinition method, XPathNodeIterator iterator)
+		{
+			var methodParameters = new ArrayBuilder<ArrayBuilder<Attribute>> ();
+
+			while (iterator.MoveNext ()) {
+				methodParameters.Add (ProcessAttributes (iterator.Current.SelectChildren ("attribute", _ns)));
+			}
+			return methodParameters;
+		}
+
+
+		void ProcessMethodIfNotNull (TypeDefinition type, MethodDefinition method, XPathNodeIterator iterator)
+		{
+			if (method == null)
+				return;
+
+			ProcessMethodChildren (type, method, iterator);
+		}
+
+		void ProcessMethodName (TypeDefinition type, string name, XPathNodeIterator iterator)
+		{
+			if (!type.HasMethods)
+				return;
+
+			foreach (MethodDefinition method in type.Methods)
+				if (name == method.Name)
+					ProcessMethodChildren (type, method, iterator);
+		}
+
+		protected static MethodDefinition GetMethod (TypeDefinition type, string signature)
+		{
+			if (type.HasMethods)
+				foreach (MethodDefinition meth in type.Methods)
+					if (signature == GetMethodSignature (meth, false))
```

Comment:
This won't match if there's e.g. a whitespace difference. It would be nice if it matched.

Reply:
Now it will match no matter the whitespaces